core/include/assert.h

```diff
@@ -47,6 +47,19 @@ extern "C" {
 #endif
 
 /**
+ * @brief   Filename macro
+ *
+ * @note    This was introduced to allow external packages that don't use RIOT's
+ *          build system to also use the assert macro defined via @ref
+ *          DEBUG_ASSERT_VERBOSE without requiring the `RIOT_FILE_RELATIVE`.
+ */
+#ifdef RIOT_FILE_RELATIVE
+#define DEBUG_ASSERT_FILENAME   (RIOT_FILE_RELATIVE)
```

Comment:
I actually would tend to define `RIOT_FILE_RELATIVE` as `__FILE__` if not defined (as default). Otherwise, we would have yet another macro variable ..